src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java

```diff
@@ -258,6 +255,11 @@ default User getUserWithGroupsAndAuthorities(@NotNull String username) {
         return findOneWithGroupsAndAuthoritiesByLogin(login);
     }
 
+    @NotNull
+    default User findWithGroupsAndAuthoritiesByIdElseThrow(long userId) {
```

Comment:
In regard to consistency I would propose `findOneWith...`

Reply:
We actually would like to unify this with other repositories, so it should be `findByIdWith...ElseThrow`

All the methods with `findOne...` should be renamed accordingly